lib/lighthouse.js

```diff
@@ -89,20 +104,47 @@ function runTestsForUrl({ url, shouldSignIn, hide }) {
           const erroredElements = getErroredElements({ details, extendedInfo, hide });
 
           if (erroredElements.length > 0) {
-            const errorDetail = getErrorDetail(erroredElements);
-            testCase.failure(errorMessage + errorDetail);
+            logErrorDetailsAsTestFailures(testCase, erroredElements, errorMessage);
           }
         }
       });
+      return results;
+    })
+    .then((results) => {
+      const scores = getScoresPerCategory(results.reportCategories, url);
+      jsonOutput.urls.push(scores);
     })
     .catch((err) => {
       logger.error(err);
       process.exit(1);
     });
 }
 
-function launchChromeAndRunLighthouse(url, shouldSignIn) {
-  const { flags: lighthouseFlags, config: lightHouseConfig } = LIGHTHOUSE_OPTS;
+function getLighthouseConfig(config) {
+  const lighthouseConfig = getDefaultLighthouseConfig();
+
+  if (config.hasOwnProperty('lighthouseCategories')) {
+    lighthouseConfig.config.settings.onlyCategories =
+      config.lighthouseCategories;
+  }
+
+  return lighthouseConfig;
+}
+
+function getOutputJson() {
```

Comment:
This name would be more accurate as `getOutputJSONPath` as it gets the path, not the JSON